### PR TITLE
MySQL databases - service naming

### PR DESCRIPTION
**Pull request by jbertran**
### What does this PR do?

This applies the service naming schema logic (see #2941) to mysql integrations - `mysql`, `mysql2` and `mariadb`.

These are more complex than messaging integrations, as we need to take into account the default provided by DatabasePlugin into the naming functions, which version 0 _must_ respect, but version 1 does not need.

This PR relies on #3056 for the split into different files according to general area (messaging, storage) and a few optimizations to the service naming computation.

### Motivation

Keep trucking on naming schema adoption.

### Plugin Checklist

- [x] Unit tests.

### CODE REVIEW

**Review by tlhunter — DISMISSED** [review]

File: packages/datadog-plugin-mariadb/test/naming.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { resolveNaming } = require('../../dd-trace/test/plugins/helpers')
+
+module.exports = resolveNaming({
+  outbound:
+  { v0: {
```
Comment (rochdev) [inline packages/datadog-plugin-mariadb/test/naming.js:5]: The indentation of this whole object is really hard to follow.

Comment (rochdev) [reply packages/datadog-plugin-mariadb/test/naming.js:5]: Looks like the indentation for `mysql` is fine, let's use that here too.

File: packages/datadog-plugin-mysql2/test/naming.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const { resolveNaming } = require('../../dd-trace/test/plugins/helpers')
+
+module.exports = resolveNaming({
+  outbound:
```
Comment (rochdev) [inline packages/datadog-plugin-mysql2/test/naming.js:4]: Same as above 😕

**Review by rochdev — DISMISSED** [review]

File: packages/datadog-plugin-mariadb/test/naming.js (hunk 1 above)
Comment (Qard) [reply packages/datadog-plugin-mariadb/test/naming.js:5]: A bit surprised eslint doesn't complain about that. 🤔

Comment (jbertran) [reply packages/datadog-plugin-mariadb/test/naming.js:5]: Oof, that one sneaked out - these got refactored, indeed it's surprising there's no warning on it.

**Review by tlhunter — APPROVED** [review]

**Review by tlhunter — APPROVED** [review]
